metadata-ingestion/src/datahub/specific/dataproduct.py

```diff
@@ -152,7 +152,7 @@ def set_external_url(self, external_url: str) -> "DataProductPatchBuilder":
         self._add_patch(
             DataProductProperties.ASPECT_NAME,
             "replace",
```

Comment:
This should be add

Reply:
Why? I am not sure about the semantics here. But in case it does exist `replace` sounds correct, no?

Reply:
https://jsonpatch.com/  In case we are just adding sure but if we are replacing then `replace` sounds fine. Would `add` work in case it already exists? Let me try that actually.

Reply:
I tried both `add` and `replace` and both work in our case. I am not sure whether there is a semantic reason to use either one. I'll merge this for now but if there is a semantic reason to use `add` can send a follow-up once I understand the reasoning for the use of `add` over `replace`.

Reply:
It's because of the spec, add and replace are not intuitive. Add is an upsert and replace is really a "modifyIfExists" and throws an error if the value is not there. I think maybe they modify it in the python side to be add because I specifically did not include replace as a supported operation.

Reply:
It's fine, I was planning on changing all of them with my latest PR anyway.